File: Server/src/Server.Application/ViewModels/AddressViewModel.cs
namespace Server.Application.ViewModels;

public class AddressViewModel
{
    public string PostalCode { get; set; }
    public string City { get; set; }
    public string Street { get; set; }
    public string Number { get; set; }
    public string? Apartment { get; set; }
    public string? AdditionalInfo { get; set; }
}

File: client/src/models/AddressViewModel.ts
import { z } from "zod";

const AddressViewModel = z.object({
    postalCode: z.string(),
    city: z.string(),
    street: z.string(),
    number: z.string(),
    apartment: z.string().nullable(),
    additionalInfo: z.string().nullable(),
})

export default AddressViewModel;
export type AddressViewModelType = z.infer<typeof AddressViewModel>;